commit f20df2eb8545f2a4fc16acdde5edcb611868522e
Author: Patrick Donnelly <pdonnell@ibm.com>
Date:   Thu Jan 30 17:08:07 2025 -0500

    mgr: excise CephFS client from mgr C++ base
    
    Linking to the client causes two copies of the Client library to be linked in
    the ceph-mgr when modules also dynamically link to libcephfs via the "cephfs"
    python library. This creates problems with duplicate boost::locale.
    
    Instead, modules should just use the "cephfs" library to send commands to the
    MDS.
    
    Signed-off-by: Patrick Donnelly <pdonnell@ibm.com>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index 052e6868177..3fc14d9eb29 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -52,7 +52,6 @@ MgrStandby::MgrStandby(int argc, const char **argv) :
 		     "mgr",
 		     Messenger::get_random_nonce())),
   objecter{g_ceph_context, client_messenger.get(), &monc, poolctx},
-  client{client_messenger.get(), &monc, &objecter},
   mgrc(g_ceph_context, client_messenger.get(), &monc.monmap),
   log_client(g_ceph_context, client_messenger.get(), &monc.monmap, LogClient::NO_FLAGS),
   clog(log_client.create_channel(CLOG_CHANNEL_CLUSTER)),
@@ -131,7 +130,6 @@ int MgrStandby::init()
   // Initialize Messenger
   client_messenger->add_dispatcher_tail(this);
   client_messenger->add_dispatcher_head(&objecter);
-  client_messenger->add_dispatcher_tail(&client);
   client_messenger->start();
 
   poolctx.start(2);
@@ -198,7 +196,6 @@ int MgrStandby::init()
   objecter.set_client_incarnation(0);
   objecter.init();
   objecter.start();
-  client.init();
   timer.init();
 
   py_module_registry.init();
@@ -369,7 +366,7 @@ void MgrStandby::handle_mgr_map(ref_t<MMgrMap> mmap)
       dout(1) << "Activating!" << dendl;
       active_mgr.reset(new Mgr(&monc, map, &py_module_registry,
                                client_messenger.get(), &objecter,
-			       &client, clog, audit_clog));
+			       clog, audit_clog));
       active_mgr->background_init(new LambdaContext(
             [this](int r){
               // Advertise our active-ness ASAP instead of waiting for
diff --git a/src/mgr/MgrStandby.h b/src/mgr/MgrStandby.h
index 5d238c85577..35ce43c0c80 100644
--- a/src/mgr/MgrStandby.h
+++ b/src/mgr/MgrStandby.h
@@ -21,7 +21,6 @@
 #include "common/Timer.h"
 #include "common/LogClient.h"
 
-#include "client/Client.h"
 #include "mon/MonClient.h"
 #include "osdc/Objecter.h"
 #include "PyModuleRegistry.h"
@@ -44,7 +43,6 @@ protected:
   MonClient monc;
   std::unique_ptr<Messenger> client_messenger;
   Objecter objecter;
-  Client client;
 
   MgrClient mgrc;
 
